lib/chef/resource/openssl_x509_certificate.rb

```diff
@@ -109,30 +109,41 @@ class OpensslX509Certificate < Chef::Resource
       property :ca_key_pass, String,
         description: "The passphrase for CA private key's passphrase."
 
+      property :renew_before_expiry, Integer,
+        description: "The number of days before the expiry. The certificate will be automaticaly renewed when the value is reached."
+
       action :create do
         description "Generate a certificate"
 
-        unless ::File.exist? new_resource.path
-          converge_by("Create #{@new_resource}") do
-            file new_resource.path do
-              action :create_if_missing
-              owner new_resource.owner unless new_resource.owner.nil?
-              group new_resource.group unless new_resource.group.nil?
-              mode new_resource.mode unless new_resource.mode.nil?
-              sensitive true
-              content cert.to_pem
-            end
-
-            if new_resource.csr_file.nil?
-              file new_resource.key_file do
-                action :create_if_missing
-                owner new_resource.owner unless new_resource.owner.nil?
-                group new_resource.group unless new_resource.group.nil?
-                mode new_resource.mode unless new_resource.mode.nil?
-                sensitive true
-                content key.to_pem
-              end
-            end
+        file new_resource.path do
+          action :create_if_missing
+          owner new_resource.owner unless new_resource.owner.nil?
+          group new_resource.group unless new_resource.group.nil?
+          mode new_resource.mode unless new_resource.mode.nil?
+          sensitive true
+          content cert.to_pem
+        end
+
+        unless new_resource.renew_before_expiry.nil?
```

Comment:
We should do this all inline so we don't have a resource appear in the log that we skip:

```ruby
if !new_resource.renew_before_expiry.nil? && cert_need_renewall?(new_resource.path, new_resource.renew_before_expiry)
```

Reply:
Done.